nestkernel/node_collection.cpp

```diff
@@ -413,17 +422,14 @@ NodeCollectionPrimitive::NodeCollectionPrimitive::slice( size_t start, size_t st
   }
 
   NodeCollectionPTR sliced_nc;
-  if ( step == 1 )
+  if ( step == 1 and metadata_ == nullptr )
   {
-    sliced_nc = std::make_shared< NodeCollectionPrimitive >( first_ + start, first_ + stop - 1, model_id_, metadata_ );
+    // Subtract one from the last ID because end is one past the last node.
```

Comment:
```suggestion
    // Create primitive NodeCollection passing node IDs.
    // Subtract 1 because "end" is one past last element to take while constructor expects ID of last node.
```